AggregateMetrics/AggregateMetrics/AzureWebApp/AzureWebApp.cs

```diff
@@ -0,0 +1,65 @@
+﻿using Microsoft.ApplicationInsights.Extensibility.AggregateMetrics.Two;
+using System;
+using Microsoft.ApplicationInsights.DataContracts;
+using System.Net.Http;
+using System.Runtime.Caching; 
+
+namespace Microsoft.ApplicationInsights.Extensibility.AggregateMetrics.AzureWebApp
+{
+
+    class PerformanceCounterGauge : ICounterValue
+
+    {
+        public string GetJson()
+        {
+            HttpClient client = new HttpClient();
+            var t = client.GetStringAsync("http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_APP/");
+            t.Wait();
+
+            return t.Result;
+        }
+
+        public string GetValue(string json)
+        {
+            string prefix = "\\\"privateBytes\\\":";
+            string postfix = ",\\";
+
+            int idx = json.IndexOf(prefix) + prefix.Length;
+            int endIdx = json.IndexOf(postfix, idx);
+
+            return json.Substring(idx, endIdx - idx);
+        }
+
+        public MetricTelemetry GetValueAndReset()
+        {
+            ObjectCache cache = MemoryCache.Default;
+            CacheItemPolicy policy = new CacheItemPolicy();
+            policy.AbsoluteExpiration = DateTimeOffset.Now.AddSeconds(5.0);
+
+            //http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_ASPNET/
+            //http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_APP/
+            //http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_CLR/
+            //http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_ALL/
+
+            if (cache["jsonContent"] == null)
+            {
+                string json = GetJson();
+                cache.Set("jsonContent", json, DateTimeOffset.Now.AddSeconds(5.0));
+            }
+
+            string cachedJson = (string)cache.Get("jsonContent");
+
+            string value = GetValue(cachedJson);
```

Comment:
While it is highly unlikely, there is a theoretical possibility for the cache.Set to occur, then the thread get interrupted and the CPU get maxed and used for something else for 5 seconds, and then the cache.Get to return null (really unlikely though).
It's been a while since I used the MemoryCache class but isn't there a way to wire it up to call a user defined population function when a value is requested which is not found in the cache?  Essentially when you do the cache.Get call it will internall call your predefined population function if it does not have an up to date version of what you asked for.  If it does not already have that functionality then I must have it in some old code from another project I can find and give you.